cpp/src/Slice/Scanner.l

```diff
@@ -61,144 +39,430 @@ namespace Slice
 {
 
 // Definitions for the case-insensitive keyword-token map.
-typedef std::map<std::string, int> StringTokenMap;
-static StringTokenMap keywordMap;
+static map<string, int> keywordMap;
 
-void initScanner();
-int checkKeyword(string&);
-int checkIdentifier(string&);
+static int checkKeyword(string&);
+static int checkIdentifier(const string&);
 
 }
 
+// Stores the scanner's current column position. Flex also automatically
+// generates 'yylineno', which stores the scanner's current line number.
+int yycolno = 0;
+// Stores a copy of the filename that the scanner is currently scanning.
+shared_ptr<string> yyfilename;
+
+static void nextLine(int = 1);
+static int scanPosition(const char*);
+static void setLocation(TokenContext*);
+static void startLocation(TokenContext*);
+static void endLocation(TokenContext*);
+
+static void initScanner();
+static void preAction();
+static void yynoreturn fatalError(const char* msg);
+
+// Override some of the functions flex auto-generates with our own implementations.
 #define YY_USER_INIT initScanner();
+#define YY_USER_ACTION preAction();
+#define YY_FATAL_ERROR(msg) fatalError(msg);
 
 %}
 
-%option noyywrap
-%option never-interactive
+  /* Changes the default prefix of 'yy' to 'slice_' for functions and variables in the generated code. */
```

Comment:
Does flex support C++ style comments?

Reply:
It's supposed to, but apparently there's been lots of issues with them being parsed correctly. Even in the manual itself, it recommends using C-style comments that are indented by at least 1 whitespace character.